src/TLuaInterpreter.cpp

```diff
@@ -17923,3 +17925,75 @@ int TLuaInterpreter::setMapRoomExitsColor(lua_State* L)
     lua_pushboolean(L, true);
     return 1;
 }
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#prompt
+int TLuaInterpreter::prompt(lua_State *L)
+{
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "prompt: bad argument #1 type (dialog title as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    if (!lua_isstring(L, 2)) {
+        lua_pushfstring(L, "prompt: bad argument #2 type (dialog label as string expected, got %s!)", luaL_typename(L, 2));
+        return lua_error(L);
+    }
+
+    if (!lua_isfunction(L, 3)) {
+        lua_pushfstring(L, "prompt: bad argument #3 type (callback as function expected, got %s!)", luaL_typename(L, 3));
+        return lua_error(L);
+    }
+
+    if (lua_gettop(L) >= 4 && !lua_istable(L, 4)) {
+        lua_pushfstring(L, "prompt: bad argument #4 type (option list as table expected, got %s!)", luaL_typename(L, 4));
+        return lua_error(L);
+    }
+
+    auto *dialog = new QInputDialog(mudlet::self(), Qt::Dialog);
+    dialog->setWindowTitle(lua_tostring(L, 1));
+    dialog->setLabelText(lua_tostring(L, 2));
+    dialog->setOkButtonText(tr("OK"));
+    dialog->setCancelButtonText(tr("Cancel"));
+    dialog->setFixedSize(mudlet::self()->width() / 3, 70);
+
+    if (lua_istable(L, 4)) {
+        QStringList options;
+        lua_pushnil(L);
+        while (lua_next(L, 4) != 0) {
+            if (lua_type(L, -1) == LUA_TSTRING) {
+                QString cmd = lua_tostring(L, -1);
+                options << cmd;
+            }
+            lua_pop(L, 1);
+        }
+        dialog->setComboBoxItems(options);
+    }
+
+    lua_pushvalue(L, 3);
+    int callback = lua_ref(L, LUA_REGISTRYINDEX);
+
+    connect(dialog, &QDialog::accepted, [=]() {
+        lua_rawgeti(L, LUA_REGISTRYINDEX, callback);
+        lua_pushstring(L, dialog->textValue().toUtf8());
+        int error = lua_pcall(L, 1, 0, 0);
+        if (error != 0) {
```

Comment:
:neckbeard: the `!= 0` is unnecessary here - in C/C++ any integer that is NOT `0` is logically `true`...! {Lua is different but this isn't Lua, :laughing: }

Reply:
Oh OK, actually, copied that one from somewhere else, but good to know :)